types/transaction.go

```diff
@@ -61,6 +61,7 @@ type Transaction struct {
 	V, R, S   *big.Int
 	Hash      Hash
 	From      Address
+	ChainId   *big.Int
```

Comment:
In order to keep linter happy :slightly_smiling_face: 
```suggestion
	ChainID   *big.Int
```

Where this field is even populated and used? Could it be the reason why unit tests are failing (https://github.com/0xPolygon/polygon-edge/actions/runs/5581833571/jobs/10235267467?pr=1729)?